test/solc/CommandLineParser.cpp

```diff
@@ -436,20 +434,29 @@ BOOST_AUTO_TEST_CASE(experimental_via_ir_invalid_input_modes)
 		"--standard-json",
 		"--link",
 	};
-	for (string const& inputModeOption: inputModeOptions)
+
+	static array<string, 2> const invalidOptions = {
+		"--error-recovery",
+		"--experimental-via-ir",
+	};
```

Comment:
It will work like this in the current PR but only because both options are supported in the same modes. To handle options from different modes you'll later need to store option-mode combinations so you could already do it here.